core/src/dsnp/api_types.rs

```diff
@@ -29,6 +31,7 @@ pub struct PageData {
 
 /// implementing input validation for Page Data
 impl InputValidation for PageData {
+	#[log_result(Level::Info)]
```

Comment:
I think macro name is a bit misleading since you are only logging the errors not result, so maybe this should be renamed to `log_result_error` or `log_error_result`

Reply:
Good point. I was also thinking about maybe extending the macro to be able to log a `Result::Ok` at a different log level (say, `Debug` than the `Err`). Something like `#[log_result(Level::Error, Some(Level::Debug))]` or `#[log_result(Level::Error, None)]`

Reply:
Thoughts on:
1. Rename macro
or
2. Add support for logging the `Result::Ok` as well + add as `Level::Debug` for some lower-level functions (small amount of work)

Reply:
I think logging the result even though is useful it would restrict the return types too much, since they would need to implement Debug or Display. I think for return result logging we can always log them manually but open to other perspectives also

Reply:
Renamed macro.